Fix #233 Add limit pushdown to attached queries

noahisaksen opened the pull request:
Implements limit pushdown https://github.com/duckdb/duckdb-postgres/issues/233

Gets applied on all postgres attached queries, if preferred can be applied only when an option like experimental pushdown is set, however in this implementation triggers the same as the mysql_scanner limit pushdown implementation. 

This change greatly improves performance for queries that include a `LIMIT`. 

What it does:

`select bar from attached_pg.foo limit 1;`

Previously

`COPY (SELECT "bar" FROM "attached_pg"."foo" ) TO STDOUT (FORMAT "binary");`

Now

`COPY (SELECT "bar" FROM "attached_pg"."foo"  LIMIT 1) TO STDOUT (FORMAT "binary");`

------

I am not sure how a specific test for this would look like as I am not sure there is a way to verify generated SQL. A generic limit test was added. 

------

Works for `LIMIT` and `OFFSET`. If an `ORDER_BY` is present it will not be added.

Mytherin commented:
Thanks for the PR!

Could you just fix the merge conflict?

As for testing, you can use the `EXPLAIN` to verify the limit is no longer part of the DuckDB plan, e.g.:

```sql
statement ok
set explain_output='optimized_only'

query II
EXPLAIN FROM s.large_tbl LIMIT 5;
----
logical_opt	<!REGEX>:.*"name": "LOGICAL_LIMIT".*
```

noahisaksen commented:
@Mytherin Thanks for the tip on the Query Plan test! 

Fixed the merge conflict and added a test to verify the LIMIT is not inside the DuckDB plan.

Mytherin commented:
Thanks!
